cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -614,7 +601,37 @@ private Path getProperInstallationSubDirOf(Path path, Path archiveFile) {
   @Override
   public void extractZip(Path file, Path targetDir) {
 
-    extractZipArchive(file, targetDir);
+    this.context.info("Extracting ZIP file {} to {}", file, targetDir);
+    URI uri = URI.create("jar:" + file.toUri());
+    try (FileSystem fs = FileSystems.newFileSystem(uri, FS_ENV)) {
+      long size = 0;
+      for (Path root : fs.getRootDirectories()) {
+        size += getFileSizeRecursive(root);
+      }
+      try (IdeProgressBar bp = this.context.newProgressbarForExtracting(size)) {
+        PathCopyListener listener = (source, target, directory) -> {
```

Comment:
Can you please explain this in our next KT-Session?

Reply:
Sure, but I already did or tried to do in our last "Sprint Review" meeting but maybe we have been to short on time so lets repeat that tomorrow.